src/Analyzers/CSharp/Analyzers/RemoveUnnecessarySuppressions/CSharpRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Reflection;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.RemoveUnnecessarySuppressions;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessarySuppressions
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer
+        : AbstractRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer
+    {
+        protected override string CompilerErrorCodePrefix => "CS";
+        protected override int CompilerErrorCodeDigitCount => 4;
+        protected override ISyntaxFacts SyntaxFacts => CSharpSyntaxFacts.Instance;
+        protected override (Assembly assembly, string typeName) GetCompilerDiagnosticAnalyzerInfo()
+            => (typeof(SyntaxKind).Assembly, CompilerDiagnosticAnalyzerNames.CSharpCompilerAnalyzerTypeName);
```

Comment:
slightly worried to see Assembly here... will hold off judgemnt to see how this is used.